shark_turbine/kernel/wave/wave_sim.py

```diff
@@ -33,12 +35,16 @@ def decorator(f: Callable[..., Any]) -> Callable[..., Any]:
 
         def func_wrapper(*args):
             global _api_subs
+            global _symbolic_shapes
             subs = copy.copy(_api_subs)
             if args_handler:
                 args_handler(args, subs)
 
             new_func = _resolve_symbols(f, subs)
-            return new_func(*args)
+            try:
+                return new_func(*args)

```

Comment:
Why would returning new_func(*args) fail?

Reply:
We are calling `new_func` here, so I want cleanup to happen even if it fails for whatever reason.